### PR TITLE
Fix atom masses in vanilla HybridTopologyFactory

### LINKED ISSUE #910: Simple neq switching example has nans
When I run neq switching (with the vanilla htf) for ALA -> THR (dipeptide) in solvent, I am seeing nans toward the end of forward switching. Note that the nans do not occur every time I run the experiment. If I run 3x, I see nans in 2/3 runs. 

Also note that this only happens with master. If i run with perses=0.9.2 or 0.9.1, I do not see any nans. 

All experiments run with openmm = 7.7.0 from conda-forge (not the nightly dev builds).

Code:
```python
from perses.tests.test_topology_proposal import generate_atp, generate_dipeptide_top_pos_sys
import pickle

# Generate vanilla htf
atp, system_generator = generate_atp(phase='solvent')
htf = generate_dipeptide_top_pos_sys(atp.topology,
                                     'THR',
                                     atp.system,
                                     atp.positions,
                                     system_generator,
                                     conduct_htf_prop=True,
                                     validate_endstate_energy=False)

with open("atp_solvent_vanilla.pickle", "wb") as f:
    pickle.dump(htf, f)

import numpy as np
from openmmtools.integrators import PeriodicNonequilibriumIntegrator
from openmmtools.constants import kB
from simtk import unit, openmm
import argparse
import os
import time

# Define simulation parameters
nsteps_eq = 25000 # 100 ps
nsteps_neq = int(250000) # 1 ns
neq_splitting='V R H O R V'
timestep = 4.0 * unit.femtosecond
platform_name = 'CUDA'
temperature = 300.0 * unit.kelvin

# Define lambda functions
x = 'lambda'
ALCHEMICAL_FUNCTIONS = {
                             'lambda_sterics_core': x,
                             'lambda_electrostatics_core': x,
                             'lambda_sterics_insert': f"select(step({x} - 0.5), 1.0, 2.0 * {x})",
                             'lambda_sterics_delete': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
                             'lambda_electrostatics_insert': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
                             'lambda_electrostatics_delete': f"select(step({x} - 0.5), 1.0, 2.0 * {x})",
                             'lambda_bonds': x,
                             'lambda_angles': x,
                             'lambda_torsions': x}

# Read in vanilla htf
print("reading htf")
with open(os.path.join("atp_solvent_vanilla.pickle"), 'rb') as f:
    htf = pickle.load(f)
    
positions = htf.hybrid_positions
system = htf.hybrid_system

# Set up integrator
print('setting up integrator')
integrator = PeriodicNonequilibriumIntegrator(ALCHEMICAL_FUNCTIONS, nsteps_eq, nsteps_neq, neq_splitting, timestep=timestep, temperature=temperature)

# Set up context
print("setting up context")
platform = openmm.Platform.getPlatformByName(platform_name)
if platform_name in ['CUDA', 'OpenCL']:
    platform.setPropertyDefaultValue('Precision', 'mixed')
if platform_name in ['CUDA']:
    platform.setPropertyDefaultValue('DeterministicForces', 'true')
context = openmm.Context(system, integrator, platform)
context.setPeriodicBoxVectors(*htf.hybrid_system.getDefaultPeriodicBoxVectors())
context.setPositions(positions)
context.setVelocitiesToTemperature(temperature)

# Minimize
print("minimize")
openmm.LocalEnergyMinimizer.minimize(context)

# Run eq forward (0 -> 1)
print("running eq")
integrator.step(nsteps_eq)

# Run neq forward (0 -> 1)
print("running neq")
energies = []
positions_old = []
positions_new = []
for fwd_step in range(int(nsteps_neq / 2500)):
    integrator.step(2500)
    print(f"Forward neq: {fwd_step*2500} completed")

```

### PR DESCRIPTION
**Pull request by zhang-ivy**
## Description
Takes the average of old and new system masses for mapped (environment/core) atoms.

## Motivation and context

Currently, the `HybridTopologyFactory` adds all environment/core/unique old atoms to the hybrid system using the mass of the particle in the old system. It then adds unique new atoms using the mass of the particle in the new system.

To improve stability, we want to average the old and new system masses for any mapped atom (environment/core atoms). 

Resolves #910

## Change log

```

```

### CODE REVIEW

File: perses/annihilation/relative.py [hunk 1]
```diff
@@ -2527,9 +2534,6 @@ def _handle_nonbonded(self):
         hybrid_to_old_map = self._hybrid_to_old_map
         hybrid_to_new_map = self._hybrid_to_new_map
 
-        #create the forces...
-        self._hybrid_system_forces['NonbondedForce'] = openmm.NonbondedForce()
```
Comment (ijpulidos) [inline perses/annihilation/relative.py:2531]: Why is this not needed anymore? Was it ever needed?

File: perses/annihilation/relative.py [hunk 2]
```diff
@@ -237,13 +237,20 @@ def __init__(self,
...
         for particle_idx in range(self._topology_proposal.n_atoms_old):
-            particle_mass = self._old_system.getParticleMass(particle_idx)
+            particle_mass_old = self._old_system.getParticleMass(particle_idx)
+            
+            if particle_idx in self._topology_proposal.old_to_new_atom_map.keys():
+                particle_index_in_new_system = self._topology_proposal.old_to_new_atom_map[particle_idx]
+                particle_mass_new = self._new_system.getParticleMass(particle_index_in_new_system)
+                particle_mass = (particle_mass_old + particle_mass_new) / 2 # Take the average of the masses if the atom is mapped
+            else:
+                particle_mass = particle_mass_old
```
Comment (ijpulidos) [inline perses/annihilation/relative.py:247]: This looks fine to me. I have the feeling one could simplify the logic here, specially considering the if statement in line 253. But I guess this is easy enough to follow as it is and the possible performance gained by it is not critical, so this is not a blocker. Thanks!!

**Review by ijpulidos — APPROVED** [review]
Looks good to me!! Thanks!

File: perses/annihilation/relative.py (hunk 1 above)
Comment (zhang-ivy) [reply perses/annihilation/relative.py:2531]: If you search for `self._hybrid_system_forces['NonbondedForce']` in the file, its never used anywhere. I don't think it was ever needed.

**Pull request merged**